PBN RCT.Rmd

```diff
@@ -27,157 +27,210 @@ if(!require(openxlsx)){install.packages('openxlsx')}
+#Compare to a dataframe of folks who completed all of the PFS items
+check_pfs_all <- included_participants %>%
   mutate(pi_pfs_sum = pi_pfs_1 + pi_pfs_2 + pi_pfs_3 + pi_pfs_4 + pi_pfs_5 + pi_pfs_6 + pi_pfs_7) %>%
   filter(!is.na(pi_pfs_sum))
+#Identify the one different participant ID:
+missing_participant_ids <- setdiff(check_pfs_partial$Participant_ID, check_pfs_all$Participant_ID)
+print(missing_participant_ids)
+#The participant with missing data is 64531 
+###FIGURE OUT WHAT TO DO WITH THEM###
+
+#Create score and filter for program feedback completers (just renaming for later code here since the data frame was made above)
+included_participants_pfs <- check_pfs_all
 
 #Identify participants with greater than 5% missing baseline primary outcomes data
-#For each participant, add a new column that calculates the total number of baseline primary outcome items answered by each participant 
-included_participants <- included_participants %>%
-  mutate(total_responses_bprimary = rowSums(
-    !is.na(select(included_participants, b_mfq_1, b_mfq_2, b_mfq_3, b_mfq_4, b_mfq_5, b_mfq_6, b_mfq_7, b_mfq_8, b_mfq_9, b_mfq_10, b_mfq_11, b_mfq_12, b_mfq_13, b_edeq_1, b_edeq_2, b_edeq_3, b_edeq_4, b_edeq_5, b_edeq_6, b_edeq_7, b_edeq_8, b_edeq_9, b_edeq_10)), 
-    na.rm = TRUE))
-#For each participant, calculate the percent of missing responses (there are 23 items total)
+#For each participant, add a new column that calculates the percent missing of baseline primary outcome items
 included_participants <- included_participants %>%
-   mutate(percent_missing = (1 - (total_responses_bprimary / 23)) * 100) 
+  mutate(percent_missing = rowMeans(
+    is.na(select(included_participants, b_mfq_1, b_mfq_2, b_mfq_3, b_mfq_4, b_mfq_5, b_mfq_6, b_mfq_7, b_mfq_8, b_mfq_9, b_mfq_10, b_mfq_11, b_mfq_12, b_mfq_13, b_edeq_1, b_edeq_2, b_edeq_3, b_edeq_4, b_edeq_5, b_edeq_6, b_edeq_7, b_edeq_8, b_edeq_9, b_edeq_10)), 
+    na.rm = TRUE)) %>%
+  mutate(percent_missing = percent_missing * 100)
+
 #Make a new variable that identifies participants with more than 5% missing data (those with more than 5% missing are TRUE)
 included_participants <- included_participants %>%
   mutate(more_than_5_percent_missing = percent_missing > 5)
 #Remove participants where more_than_5_percent_missing is TRUE
 included_participants <- included_participants %>%
   filter(!more_than_5_percent_missing)
 #In sum, there were originally 220 included_participants and now there are 218 (only had to drop two participants prior to imputation)
+###NEED TO CHECK### - I think included_participants can now be considered analysis-ready for imputation (since it dropped participants with more than 5% missing baseline data and includes all the variables that I think will be useful in imputation). There are a few extra variables that I've used along the way such as Participant_ID, treatment_group, percent_missing, and more_than_5_percent missing. Keep or drop these for imputation?
```

Comment:
It's important to keep some indicator of treatment assignment for imputation, since that is a big predictor of follow-up outcomes!

Otherwise, those don't seem useful